### PR TITLE
add loglog plot and fix type errors in examples

### PR DESCRIPTION
**Pull request by jzstark**
Here are the changes:

1. Add a loglog plot. One example can be viewed [here(plot_025.png)](https://gist.github.com/jzstark/dec579b428c5a0109e3900d80117b627). 

2. Two examples (test_mnist.ml and test_neural.ml) got type inconsistency problems. According to [commit 09b6c](https://github.com/ryanrhymes/owl/commit/09b6c52fd659f578d21b065d2932aab6df4c247f), I assume it's safe to change `Owl_dense_matrix_s` to `Owl_dense_matrix.S` after the update of API. So far both examples run correctly and `owl` compiles without error. 

3. Add some unmentioned coding style from CS3110 OCaml Style Guide.

### CODE REVIEW

File: lib/owl_plot.mli [hunk 1]
```diff
@@ -193,6 +193,11 @@ val pie : ?h:handle -> ?spec:spec list -> dsmat -> unit
   Parameters: [RGB], [Fill].
  *)
 
+val loglog : ?h:handle -> ?spec:spec list -> ?x:dsmat -> dsmat -> unit
+(** [log x y]  plots all y versus x pairs with log-log scale.
+[log y] plots data in y versus their index.
+
```
Comment (ryanrhymes) [inline lib/owl_plot.mli]: Add supported parameters, e.g., RGB, LineStyle, and etc.?

File: lib/owl_plot.ml [hunk 2]
```diff
@@ -1170,6 +1170,62 @@ let pie ?(h=_default_handle) ?(spec=[]) x =
   p.plots <- Array.append p.plots [|f|];
   if not h.holdon then output h
 
+let _range_ i j =
```
Comment (ryanrhymes) [inline lib/owl_plot.ml]: there is already one implemented, i.e., `Owl_utils.range` :)

Comment (jzstark) [inline lib/owl_plot.ml]: Thanks for pointing it out. It turns out I already used the `Owl_dense_matrix.D.linspace` and forgot to delete the initial implementation of `range` :)

File: lib/owl_plot.mli (hunk 1 above)
Comment (jzstark) [inline lib/owl_plot.mli]: Sure. Failed to follow the example of other plots. Updated now.

**Comment by ryanrhymes** [comment]
This is a good PR. Thanks Jianxin. Changing `Owl_dense_matrix_s/d/c/z` and etc. to intermediate modules may cause some type issue in some modules, thanks for fixing them. 👍

**Comment by ryanrhymes** [comment]
@jzstark maybe also consider adding `semilogx` and `semilogy` two functions, both should be straightforward based on `loglog`? :)

**Comment by jzstark** [comment]
Sure it will be done

**Pull request merged**